temporal-test-server/src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -1238,12 +1267,14 @@ private static void startWorkflow(
       a.setParentWorkflowNamespace(parentExecutionId.getNamespace());
       a.setParentWorkflowExecution(parentExecutionId.getExecution());
     }
-    HistoryEvent event =
+    HistoryEvent.Builder event =
         HistoryEvent.newBuilder()
             .setEventType(EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED)
-            .setWorkflowExecutionStartedEventAttributes(a)
-            .build();
-    ctx.addEvent(event);
+            .setWorkflowExecutionStartedEventAttributes(a);
+    if (request.getLinksCount() > 0) {
+      event.addAllLinks(request.getLinksList());
```

Comment:
We should do the same validation for these links as we do in the real server.

Reply:
I don't think we are doing any validation on these in the real server, did you have something in mind?
https://github.com/temporalio/temporal/blob/main/service/history/historybuilder/history_builder.go#L190

Reply:
I'll track.

Reply:
I forgot we don't use the Nexus link here, it's a Temporal Link.

Reply:
Still requires validation though.

Reply:
https://github.com/temporalio/temporal/pull/6625

Reply:
Added the same for the test server.